src/data/nat/prime.lean

```diff
@@ -1199,4 +1199,16 @@ lemma factors_count_eq_of_coprime_right {p a b : ℕ} (hab : coprime a b) (hpb :
   list.count p (a * b).factors = list.count p b.factors :=
 by { rw mul_comm, exact factors_count_eq_of_coprime_left (coprime_comm.mp hab) hpb }
 
+/-- If prime powers of different primes can only coincide if the power is 0 -/
+lemma prim_pow_eq_of_ne
```

Comment:
we have similar lemmas in L398 of algebra/associated.lean - maybe this is the correct generality for this? (associated.of_pow_associated_of_prime)

Reply:
Thanks for pointing that out! Are you saying I should leave `data.nat.prime` alone, and instead use `associated.of_pow_associated_of_prime` (together with `associated_iff_eq` to turn `~` to `=` on `\nat`) in my main proof?

Reply:
yeah, or maybe make this lemma saying that if two prime powers are associated then they're both equal to zero